dlt/destinations/impl/filesystem/filesystem.py

```diff
@@ -590,3 +607,74 @@ def create_table_chain_completed_followup_jobs(
                 file_name = FileStorage.get_file_name_from_file_path(table_job_paths[0])
                 jobs.append(ReferenceFollowupJob(file_name, table_job_paths))
         return jobs
+
+    def get_duckdb(self, tables: List[str]) -> DuckDBPyConnection:
+        import duckdb
+        from duckdb import InvalidInputException, IOException
+
+        # cache duckdb instance and list of created views
+        db: DuckDBPyConnection = getattr(self, "_duckdb", None)
+        existing_views: List[str] = getattr(self, "_existing_duckdb_views", [])
+
+        if not db:
+            db = duckdb.connect(":memory:")
+            db.register_filesystem(self.fs_client)
+            self._duckdb = db
+            self._existing_duckdb_views = existing_views
+
+        # create all tables in duck instance
+        for ptable in tables:
+            if ptable in existing_views:
+                continue
+            existing_views.append(ptable)
+
+            folder = self.get_table_dir(ptable)
+            files = self.list_table_files(ptable)
+            # discover tables files
+            file_type = os.path.splitext(files[0])[1][1:]
+            if file_type == "jsonl":
+                read_command = "read_json"
+            elif file_type == "parquet":
+                read_command = "read_parquet"
+            else:
+                raise AssertionError(f"Unknown filetype {file_type} for table {ptable}")
+
+            # create table
+            protocol = "" if self.is_local_filesystem else f"{self.config.protocol}://"
+            files_string = f"'{protocol}{folder}/**/*.{file_type}'"
+            create_table_sql_base = (
+                f"CREATE VIEW {ptable} AS SELECT * FROM {read_command}([{files_string}])"
+            )
+            create_table_sql_gzipped = (
+                f"CREATE VIEW {ptable} AS SELECT * FROM {read_command}([{files_string}],"
+                " compression = 'gzip')"
+            )
+            try:
+                db.sql(create_table_sql_base)
+            except (InvalidInputException, IOException):
+                # try to load non gzipped files
+                db.sql(create_table_sql_gzipped)
+
+        return db
+
+    @contextmanager
+    def get_readable_relation(
+        self, *, table: str = None, sql: str = None
+    ) -> Generator[DBApiCursor, Any, Any]:
+        from dlt.destinations.impl.duckdb.sql_client import DuckDBDBApiCursorImpl
+        import sqlglot
```

Comment:
I assume this should go in some extra, or do we just ask the user to install sqlglot if not present?

Reply:
I still do not understand why you'll not derive from `DuckdbSqlClient` and put all this code into `execute_query`? why this is good:
1. we do not refer to duckdb here. we'll be able to swap engines by swapping sql clients
2. you can move sqlglot there
3. you can rewrite `get_duckdb` to just create required relations
4. then you just call super() on `execute_query` to get the right cursor

Reply:
added bonus: filesystem now implements `sql_client` that works on every bucket and our streamlit app magically works

Reply:
yes, this works now :)